### PR TITLE
Enable `root` login and disable default `core` for rhcos

### PR DESCRIPTION
**Pull request by HuijingHei**
This is to revert part of https://github.com/teemtee/tmt/commit/ee642f68940e0bccb6554ca085d8e94964d06189,
enable `root` login and disable default username `core` for rhcos.
Refer to Colin's suggestion https://github.com/teemtee/tmt/pull/1895#issuecomment-1464060434.

### CODE REVIEW

**Comment by lukaszachy** [comment]
Could you please amend the commit message? https://tmt.readthedocs.io/en/stable/contribute.html#commits

This is a cleaner solution for sure.

 @psss we need to remember to change the release notes as this goes against the `Set default user `core` for rhcos in testcloud` description (the ee642f68 isn't fully reverted as it also makes testcloud recognize rhcos as 'coreos' type)

**Comment by HuijingHei** [comment]
> Could you please amend the commit message? https://tmt.readthedocs.io/en/stable/contribute.html#commits

Updated, could you help to review? Thanks!

File: tmt/steps/provision/testcloud.py [hunk 1]
```diff
@@ -93,6 +93,18 @@ def import_testcloud() -> None:
...
+    - name: ssh_root_login.service
+      enabled: true
+      contents: |
+        [Service]
+        Type=oneshot
+        ExecStart=/usr/bin/sed -i \
+                  "s|^PermitRootLogin no$|PermitRootLogin yes|g" \
+                  /etc/ssh/sshd_config
+        [Install]
+        WantedBy=multi-user.target
```
Comment (lukaszachy) [inline tmt/steps/provision/testcloud.py:107]: Is it guaranteed that `sshd` starts (or reloads) after this unit executes?

Comment (HuijingHei) [reply tmt/steps/provision/testcloud.py:107]: Not quite sure, how about adding `Before=sshd.service`?

**Comment by travier** [comment]
Looks like it's the simplest option for RHCOS 8.

Starting with RHEL 9 this will be better written as a drop-in file in `/etc/sshd_config.d/foo.conf`.

**Review by cgwalters — APPROVED** [review]
Looks fine to me.

**Comment by lukaszachy** [comment]
Thank you @HuijingHei  for the PR and @travier, @cgwalters for the review. 

Hopefully it will get automerged soon.

**Comment by lukaszachy** [comment]
The centos-stream-8 issue (`podman: symbol lookup error: podman: undefined symbol: seccomp_notify_fd`) isn't caused by this PR. Merging.

**Comment by travier** [comment]
Note that this won't work anymore with OCP 4.13. See https://github.com/openshift/os/pull/1251. The "fix" is to make sure the line is also un-commented after the sed call, which should be relaxed (i.e. no `^`).

**Pull request merged**